lib/modules/dosomething/dosomething_kudos/dosomething_kudos.module

```diff
@@ -311,21 +316,24 @@ function dosomething_kudos_disable_reactions($nid) {
 /**
  * Grab all Kudos data required to render the markup for a given reportback fid
  * @param  int $fid Reportback file id
+ * @param  int $total_rogue_kudos Total kudos from Rogue response (If Rogue is powering the gallery).
+ * @param  bool $liked_by_current_user Response from Rogue if the current user has reacted to the post or not.
  * @return array Values needed to render Kudos markup
  */
-function dosomething_kudos_get_kudos_data_for_fid($fid = NULL) {
+function dosomething_kudos_get_kudos_data_for_fid($fid = NULL, $total_rogue_kudos = NULL, $liked_by_current_user = NULL) {
   if (! $fid) {
     return [];
   }
-
   $kudos = [];
   $kudos['fid'] = $fid;
+  $kudos['existing_kids'] = dosomething_kudos_get_kudos_for_user_by_file($fid);
   $kudos['reaction_totals'] = [
-    'heart' => dosomething_kudos_get_total_for_file_by_term_name($fid, 'heart'),
+    'heart' =>
+        $total_rogue_kudos ? $total_rogue_kudos : dosomething_kudos_get_total_for_file_by_term_name($fid, 'heart')
   ];
-  $kudos['existing_kids'] = dosomething_kudos_get_kudos_for_user_by_file($fid);
-  $kudos['reaction_totals'] = ['heart' => dosomething_kudos_get_total_for_file_by_term_name($fid, 'heart')];
   $kudos['allowed_reactions'] = [dosomething_kudos_get_term_id_by_name('heart')];
 
+  $kudos['from_rogue'] = $liked_by_current_user ? $liked_by_current_user : FALSE;
```

Comment:
Oh, I see now. Can you add a comment with more context for where this gets used so it's a bit more clear when first reading through? It also seems like this ternary may be unnecessary.